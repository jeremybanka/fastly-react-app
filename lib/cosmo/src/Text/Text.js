/* @flow */

import type { ComponentType } from "react";
import styled from "styled-components";
import { space, typography, color, variant } from "styled-system";

// TODO
type PropsType = any;
const Text: ComponentType<PropsType> = styled("span")(
  typography,
  color,
  space,
  variant({
    prop: "purpose",
    variants: {
      success: { color: "green" },
      warning: { color: "gold" },
      danger: { color: "red" },
      muted: { color: "grays.3" },
    },
  })
);

/** @component */
export default Text;
