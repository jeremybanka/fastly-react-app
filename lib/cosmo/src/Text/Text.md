Size variants:

```jsx
<Text fontSize='xxs'>xxs text</Text>
<br />
<Text fontSize='xs'>xs text</Text>
<br />
<Text fontSize='sm'>sm text</Text>
<br />
<Text fontSize='md'>md text</Text>
<br />
<Text fontSize='lg'>lg text</Text>
<br />
<Text fontSize='xl'>xl text</Text>
<br />
<Text fontSize='xxl'>xxl text</Text>
<br />
<Text fontSize='xxxl'>xxxl text</Text>
```

Weight variants:

```jsx
<Text fontWeight='normal'>Normal text</Text>
<br />
<Text fontWeight='semibold'>Semibold text</Text>
<br />
<Text fontWeight='bold'>Bold text</Text>
```

Purpose variants:

```jsx
<Text purpose='neutral'>Neutral text</Text>
<br />
<Text purpose='success'>Success text</Text>
<br />
<Text purpose='info'>Info text</Text>
<br />
<Text purpose='warning'>Warning text</Text>
<br />
<Text purpose='danger'>Danger text</Text>
<br />
<Text purpose='muted'>Muted text</Text>
```

Styled system overrides:

```jsx
<Text color="purples.3">Neutral text</Text>
```
