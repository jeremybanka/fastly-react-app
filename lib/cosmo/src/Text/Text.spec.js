/* @flow */

import React from "react";
import { render } from "enzyme";

import Text from "./Text";

describe("Text test suite", () => {
  // it('renders with className .Text', () => {
  //   const wrapper = shallow(<Text>Hi</Text>)
  //   expect(wrapper.is('.Text')).toBe(true)
  // })

  it("renders the stored snapshot", () => {
    const component = <Text>Hi</Text>;
    expect(render(component)).toMatchSnapshot();
  });
});
