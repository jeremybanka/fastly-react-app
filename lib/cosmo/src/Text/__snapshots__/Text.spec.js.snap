// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text test suite renders the stored snapshot 1`] = `
<span
  class="Text-sc-6c1xd4-0 eVgFRd"
>
  Hi
</span>
`;
